### PR TITLE
Feat: Payment method screen 

### LINKED ISSUE #32: [FEATURE] Tela para Visualização de Métodos de Pagamento
### Descrição da Feature
Adicionar uma tela onde o usuário possa visualizar os métodos de pagamento que ele utiliza. A tela permitirá que o usuário veja uma lista de todos os métodos de pagamento cadastrados, como cartão de crédito, cartão de débito, dinheiro, cheque, pix, entre outros. Cada método de pagamento será exibido com o nome e um emoji escolhido pelo usuário para facilitar a identificação. Caso o método de pagamento já tenha sido utilizado em algum lançamento, o botão de exclusão deverá ficar bloqueado.

### Problema Resolvido
Essa feature organiza e facilita o acesso aos métodos de pagamento registrados, permitindo ao usuário visualizar rapidamente quais métodos ele tem à disposição para gerenciar suas transações. Além disso, bloqueia a exclusão de métodos que já possuem lançamentos, evitando problemas de integridade de dados e melhorando a consistência do histórico de transações.

### Solução Proposta
Implementar uma tela dedicada aos métodos de pagamento, exibindo uma lista que contenha o nome e o emoji associado a cada método. Caso o método de pagamento já tenha sido utilizado, o botão de exclusão será desativado (ou oculto), informando ao usuário que ele não pode excluir o método enquanto houver transações associadas.

### Alternativas Consideradas
- Permitir a exclusão de métodos de pagamento mesmo com lançamentos associados, mas isso pode causar inconsistências no histórico de transações e confundir o usuário.

### Impacto no Projeto
Essa feature melhora a experiência do usuário, tornando o gerenciamento de métodos de pagamento mais intuitivo e seguro. A restrição de exclusão evita problemas de integridade e ajuda a manter um histórico consistente das transações.

### Tarefas Necessárias
- [ ] Criar o layout da tela de métodos de pagamento.
- [ ] Exibir lista de métodos de pagamento com nome e emoji.
- [ ] Implementar funcionalidade para adicionar e editar métodos de pagamento.
- [ ] Verificar se há lançamentos associados antes de habilitar/desabilitar o botão de exclusão.
- [ ] Testar a usabilidade e a performance da tela.

### LINKED ISSUE #33: [FEATURE] Tela para Criação e Edição de Método de Pagamento
### Descrição da Feature
Implementar uma tela para a criação e edição de métodos de pagamento, onde o usuário possa definir o nome, selecionar um emoji representativo e escolher o tipo de pagamento (ex: cartão de crédito, cartão de débito, dinheiro, cheque, pix, etc.). Essa tela deve ser integrada com a visualização de métodos de pagamento #32 , permitindo ao usuário gerenciar seus métodos de forma completa. A feature envolve três telas principais: leitura (visualização de detalhes), adição (novo método de pagamento) e edição (alterar informações do método de pagamento).

### Problema Resolvido
Facilita o gerenciamento dos métodos de pagamento pelo usuário, permitindo a criação, edição e remoção dos métodos conforme necessário. Isso possibilita uma organização mais eficiente das transações, permitindo que o usuário defina métodos específicos para cada tipo de despesa ou ganho.

### Solução Proposta
Desenvolver um fluxo de criação e edição que permita ao usuário definir o nome do método de pagamento, escolher um emoji representativo e selecionar o tipo de pagamento. Esse fluxo deve incluir restrições para a exclusão de métodos que já foram utilizados em lançamentos, desativando o botão de exclusão nesses casos.

### Alternativas Consideradas
- Limitar os métodos de pagamento a uma lista pré-definida, mas permitir que o usuário personalize os métodos oferece uma experiência mais flexível e adaptável.

### Impacto no Projeto
Essa feature permite ao usuário ter maior flexibilidade na criação e edição de métodos de pagamento, tornando o app mais adaptável às necessidades de cada usuário e melhorando a experiência de personalização.

### Tarefas Necessárias
- [ ] Criar o layout da tela de criação e edição de método de pagamento.
- [ ] Permitir a seleção de emoji, nome e tipo de pagamento.
- [ ] Implementar a verificação de lançamentos associados antes de habilitar/desabilitar o botão de exclusão.
- [ ] Exibir mensagem de confirmação para descartar alterações ou excluir o método de pagamento.
- [ ] Testar o fluxo de criação, edição e exclusão de métodos de pagamento.

### PR DESCRIPTION
**Pull request by raquelsantoss**
### Issue
closes #32 
closes #33 

### Overview
This PR includes the creation and integration of three main views for managing payment methods within the app. These views allow users to add, edit, view details, and delete payment methods while ensuring that data consistency is maintained and appropriate validations are applied.

### Key Changes

- **PaymentMethodView**: Main screen that lists all payment methods. Users can tap on a payment method to view its details or swipe to delete, with validation to prevent deletion if transactions are associated.
  
- **DetailPaymentMethodView**: Detail view for a selected payment method. Users can view the method name and icon, with options to edit or delete. This view includes a check for associated transactions before enabling deletion.

- **EditPaymentMethodView**: Form view to add or edit a payment method. Allows selection of an icon and entry of a method name. The icon selector includes a paginated grid layout, and the save/cancel actions are validated to prevent empty fields.

### CODE REVIEW

**Comment by CaioVicenzi** [comment]
image

Instead of using lots of private state variables, it's a good practice in the MVVM architecture to use a ViewModel to store all the properties that's going to update the view as a @Published var.

image

There is some logic in the view that could be inside a ViewModel.

image

This could be a Button instead of a HStack because it would allow the payment method to be taped anywhere within the list item.

You can also make the List to be right below the header, even with one or few payment methods added, that would make the app more consistent to the prototype.

File: MoneyManagement App/View/Views/PaymentMethod/DetailPaymentMethodView.swift [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                }) {
+                    Text("Apagar Método de Pagamento")
+                        .foregroundColor(.red)
+                        .font(.system(size: 16, weight: .semibold))
+                        .padding(.vertical, 12)
+                        .frame(maxWidth: .infinity, alignment: .leading)
+                        .padding(.leading, 16)
+                }
+                .background(Color.white)
+                .alert("Excluir Método de Pagamento?", isPresented: $showDeleteAlert) {
```
Comment (RafaelOsiro) [inline MoneyManagement App/View/Views/PaymentMethod/DetailPaymentMethodView.swift:77]: @raquelsantoss let's use the same alert when we use the delete button on List. You can copy and replace here.

Captura de Tela 2024-11-10 às 21 07 39

**Review by RafaelOsiro — CHANGES_REQUESTED** [review]
1. Let's try to avoid open alert when user **add** a payment method and he didn't update nothing and when he try to cancel.
2. Let's try to avoid open alert when user **edit** a payment method and he didn't update nothing and when he try to cancel. 
3. Let's avoid to open 2 sheets. When user click in a method payment he and he can read all info of that method payment. When user click to edit you open a sheet. To avoid that, try to create a state when the form is reading and is editing and is adding.

https://github.com/user-attachments/assets/70e3aa3b-19a0-44f2-b355-9c262708f4b9


https://github.com/user-attachments/assets/d2396296-8f08-43b8-9456-82cf782025de

**Review by RafaelOsiro — CHANGES_REQUESTED** [review]
When I tried to update something and after that, the "Voltar" button is not calling the alert

https://github.com/user-attachments/assets/a4dab4d2-fd74-4306-80e2-84f073a20057


Try to use the same alert we use in list on the Edit View

Simulator Screenshot - iPhone 16 Pro Max - 2024-11-10 at 23 10 25

**Comment by raquelsantoss** [comment]
> When I tried to update something and after that, the "Voltar" button is not calling the alert
> 
>  Simulator.Screen.Recording.-.iPhone.16.Pro.Max.-.2024-11-10.at.23.08.03.mp4 
> 
> Try to use the same alert we use in list on the Edit View
> 
> Simulator Screenshot - iPhone 16 Pro Max - 2024-11-10 at 23 10 25
> 
> image

Actually, this alert should appear in the edit view, immediately after the detail page, as that’s where the user is updating the data. When returning to the detail view, they should only be in view mode.

**Review by giovannifavorin — APPROVED** [review]

**Review by RafaelOsiro — APPROVED** [review]
LGTM !

**Pull request merged**